feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,745 @@
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Device.Name()
+
+	rpActiveCpuFound := false
+	for _, cpu := range cpuCards {
+		query := gnmi.OC().Component(cpu).State()
+		timestamp := time.Now().Round(time.Second)
+		component := gnmi.Get(t, dut, query)
+		cpuParent := component.GetParent()
+		if cpuParent == "" {
+			t.Errorf("ERROR: can't find parent information for CPU card %v", component)
+		}
+
+		if cpuParent == rpActive {
+			rpActiveCpuFound = true
+			cpuUtilization := component.GetCpu().GetUtilization()
+			if cpuUtilization == nil {
+				t.Errorf("ERROR: %s %s %s Type %-20s - CPU utilization data not available",
+					timestamp, deviceName, cpu, cpuParent)
+				continue
+			}
+			averageUtilization := cpuUtilization.GetAvg()
+			if averageUtilization > cpuHighUtilization {
+				t.Errorf("ERROR: %s %s Type %-20s %-10s - Utilization: %3d%%, exceeding threshold: %3d%%",
+					timestamp, deviceName, cpu, cpuParent, averageUtilization, cpuHighUtilization)
+			} else {
+				t.Logf("INFO: %s %s Type %-20s %-10s - Utilization: %3d%%",
+					timestamp, deviceName, cpu, cpuParent, averageUtilization)
+			}
+		}
+	}
+	if !rpActiveCpuFound {
+		t.Errorf("ERROR: %s Couldn't find %s CPU card", deviceName, rpActive)
+	}
+}
+
+func TestLineCardsNoHighCPUSpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Device.Name()
+
+	if deviations.LinecardCpuUtilizationUnsupported(dut) {
+		t.Skipf("Skipping test due to deviation linecard_cpu_ultilization_unsupported")
+	}
+
+	cardList := lineCards
+	for _, cpu := range cpuCards {
+		timestamp := time.Now().Round(time.Second)
+
+		query := gnmi.OC().Component(cpu).State()
+		component := gnmi.Get(t, dut, query)
+
+		cpuParent := component.GetParent()
+		if cpuParent == "" {
+			t.Errorf("ERROR: can't find parent information for CPU card %v", component)
+		}
+
+		// If cpu card's parent is line card, check cpu ultilization.
+		if contains(cardList, cpuParent) {
+			// Remove parent from the list of check cards.
+			cardList = removeElement(cardList, cpuParent)
+			// Fetch CPU utilization data.
+			cpuUtilization := component.GetCpu().GetUtilization()
+			if cpuUtilization == nil {
+				t.Errorf("ERROR: %s %s %s Type %-20s - CPU utilization data not available",
+					timestamp, deviceName, cpu, cpuParent)
+				continue
+			}
+			averageUtilization := cpuUtilization.GetAvg()
+			if averageUtilization > cpuHighUtilization {
+				t.Errorf("ERROR: %s %s Type %-20s %-10s - Utilization: %3d%%, exceeding threshold: %3d%%",
+					timestamp, deviceName, cpu, cpuParent, averageUtilization, cpuHighUtilization)
+			} else {
+				t.Logf("INFO: %s %s Type %-20s %-10s - Utilization: %3d%%",
+					timestamp, deviceName, cpu, cpuParent, averageUtilization)
+			}
+		}
+	}
+
+	if len(cardList) > 0 {
+		t.Errorf("ERROR: Didn't find cpu card for checkCards %s", cardList)
+	}
+}
+
+func TestComponentsNoHighMemoryUtilization(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Device.Name()
```

Comment:
You can just call `dut.Name()`. No need for the ".Device". Same for all such instances below

Reply:
Thanks for your comments. Fixed it. Replace dut.Device.Name() with dut.Name(). Fixed other instances as well.
```go
func TestControllerCardsNoHighCPUSpike(t *testing.T) {
	dut := ondatra.DUT(t, "dut")
	deviceName := dut.Name()
```